src/main/java/bowling/domain/Frame.java

```diff
@@ -1,62 +1,86 @@
 package bowling.domain;
 
-import bowling.dto.FrameDto;
-
 import java.util.ArrayList;
 import java.util.List;
 
-class Frame {
+public class Frame {
     private static final int SHOT_LIMIT = 2;
 
     private final ShotScores shotScores;
-    private final boolean hasBonus;
+    private final boolean isLast;
 
-    protected Frame(List<ShotScore> shotScores, boolean hasBonus) {
-        this.shotScores = new ShotScores(shotScores);
-        this.hasBonus = hasBonus;
+    private Frame(List<ShotScore> shotScores, boolean isLast) {
+        this.shotScores = ShotScores.of(shotScores);
+        this.isLast = isLast;
     }
 
     static Frame init() {
         return new Frame(new ArrayList<>(), false);
     }
 
     Frame next(int shot) {
-        Frame frame = new Frame(new ArrayList<>(), false);
-        frame.shot(shot);
-        return frame;
+        Frame nextFrame = new Frame(new ArrayList<>(), false);
+        nextFrame.shot(shotScores.getNext(shot));
+        return nextFrame;
     }
 
     Frame last(int shot) {
-        Frame frame = new Frame(new ArrayList<>(), true);
-        frame.shot(shot);
-        return frame;
+        Frame nextFrame = new Frame(new ArrayList<>(), true);
+        nextFrame.shot(shotScores.getNext(shot));
+        return nextFrame;
     }
 
     void shot(int shot) {
-        if (shotScores.isSize(getShotLimit())) {
-            throw new IllegalStateException(String.format("shot NormalFrame fail. cannot shot over %d times", getShotLimit()));
+        shot(shotScores.getNext(shot));
+    }
+
+    private void shot(ShotScore shot) {
+        if (isShootLimitedSize()) {
+            throw new IllegalStateException(String.format("shot Frame fail. cannot shot over %d times", getShotLimit()));
         }
 
-        shotScores.add(shot, hasBonus);
+        shotScores.add(shot);
     }
 
     boolean isFrameSet() {
-        return shotScores.isSize(getShotLimit()) ||
-                (cannotShooting());
+        return isShootLimitedSize() ||
+                cannotShooting();
+    }
+
+    private boolean isShootLimitedSize() {
+        return shotScores.hasSize(getShotLimit());
+    }
+
+    public boolean isScoreCalculated() {
+        if(isLast){
```

Comment:
IntelliJ IDEA의 코드 자동 정렬 기능(⌥⌘L, Ctrl+Alt+L)을 사용하면 더 깔끔한 코드를 볼 수 있을 거예요.

```suggestion
        if (isLast) {
```